crates/driver/src/domain/competition/pre_processing.rs

```diff
@@ -394,8 +394,12 @@ impl Utilities {
         let _timer2 =
             observe::metrics::metrics().on_auction_overhead_start("driver", "cow_amm_orders");
 
-        let cow_amms = self
-            .cow_amm_cache
+        let Some(ref cow_amm_cache) = self.cow_amm_cache else {
+            // CoW AMM is not configured, return empty vec
+            return Arc::new(Default::default());
```

Comment:
Nit: Arc<T> impls Default where T impls Defaults
```suggestion
            return Default::default();
```